src/convolution/convolution_1D.jl

```diff
@@ -0,0 +1,1757 @@
+### A Pluto.jl notebook ###
+# v0.19.25
+
+#> [frontmatter]
+#> title = "Convolution"
+#> tags = ["convolution", "filter", "signal-processing"]
+#> description = "Learn about the cool concept of convolution on continuous functions!"
+
+using Markdown
+using InteractiveUtils
+
+# This Pluto notebook uses @bind for interactivity. When running this notebook outside of Pluto, the following 'mock version' of @bind gives bound variables a default value (instead of an error).
+macro bind(def, element)
+    quote
+        local iv = try Base.loaded_modules[Base.PkgId(Base.UUID("6e696c72-6542-2067-7265-42206c756150"), "AbstractPlutoDingetjes")].Bonds.initial_value catch; b -> missing; end
+        local el = $(esc(element))
+        global $(esc(def)) = Core.applicable(Base.get, el) ? Base.get(el) : iv(el)
+        el
+    end
+end
+
+# ╔═╡ 3fd4b34c-18ed-4b07-b594-01afb377ead7
+begin
+	using PlutoUI, CairoMakie, DSP, ColorSchemes, ColorTypes, SignalAnalysis, Random
+	using PlutoUI.ExperimentalLayout: grid, vbox, hbox, Div
+	using HypertextLiteral
+	using MakieThemes
+		set_theme!(ggthemr(:flat))
+end
+
+# ╔═╡ f16a932e-6667-45a8-af6f-a3a22b9cd956
+md"""
+## Convolution of functions
+Hi There! If you ever came across convolutions in some certain science context, you know they can be VERY confusing. This notebook will help you get a better intuition for convolutions. Let's go!
+"""
+
+# ╔═╡ 2c4bf3d3-b79c-4dcc-b1a8-77b9b5cd4607
+md"""Choose a smoother function and see how the results changes. Can you guess what happens when we convoluted two functions? """
+
+# ╔═╡ 1abbf55e-7f27-4c48-a0f3-f33dbf7f9ab2
+begin
+	n = 100
+	z = range(0, n, length=n)
+	fz2_select = @bind fz2_function Select(
+		[ 1 => "Gaussian", 2 => "Box"])
+	md"""Smoothing function: $(fz2_select)"""
+end
+
+# ╔═╡ 149a3384-1b45-4789-ac6b-948e7a67e296
+begin
+	e_draw = @bind e PlutoUI.Slider(10:1.0:100, default = 20)
+	md"""Index: $(e_draw)"""
+end
+
+# ╔═╡ 00358bc2-1e49-45a9-9b78-293aadd40976
+md"""## Convolution Explained
+
+It's alright if you can't tell how the resulting function is created. In the following part we will go through each step of the convolution process, to understand better how they work."""
+
+# ╔═╡ f59990f8-ecba-4d16-8b68-f3d40b044d74
+md"""Use the slider below to change the amplitude of the filter function and the range on which it's defined."""
```

Comment:
I feel like this is too complicated and amplitude modification should be after the explanation nt before